pyo3-macros-backend/src/pyclass.rs

```diff
@@ -1034,8 +1035,57 @@ fn impl_simple_enum(
         ),
         default_slots,
     )
-    .doc(doc)
-    .impl_all(ctx)?;
+    .doc(doc);
+
+    let enum_into_pyobject_impl = {
+        let output_type = if cfg!(feature = "experimental-inspect") {
+            quote!(const OUTPUT_TYPE: #pyo3_path::inspect::TypeHint = <#cls as #pyo3_path::PyTypeInfo>::TYPE_HINT;)
+        } else {
+            TokenStream::new()
+        };
```

Comment:
Looks like this bit is repeated in other functions, maybe worth refactoring into a common helper?

Reply:
I'll take a look 👍